### ISSUE TITLE
Narrator does not announce list suggestions when typing in a search box

### ISSUE THREAD

**Issue by colinmorris1992** [microsoft/fluentui#21782]
### Environment Information

- **Package version(s)**: 
- **Browser and OS versions**: (OS Build 22533.1001) Edge Dev Version: Version 99.0.1135.5 (Official build) dev (64-bit)

### Describe the issue:

With Narrator, or any screen reader, turned on there is no announcement for the To/CC/Bcc suggestions in Monarch. This is true of any list suggestions that appear after a user types in the searchbox.

### Please provide a reproduction of the issue in a codepen:
(If this is not correct, please contact colinmorris@microsoft.com)

<script src="//unpkg.com/@fluentui/react@8/dist/fluentui-react.js"></script>
<script src="//unpkg.com/@fluentui/react-hooks@8/dist/react-hooks.js"></script>
<script src="//unpkg.com/@fluentui/example-data@8/dist/example-data.js"></script>
<div id="content"></div>

#### Actual behavior:
Narrator sometimes announces the first name in the suggestion list but it never announces the second name and the user is unable to pick the suggestions with the narrator turned on

#### Expected behavior:
Pressing the down arrow through the list should focus on each item through the list. When focus is on each item, Narrator should announce the name of the person or item in the list. When the user presses enter, it should select the person or item in the list
